tasks/FileUploadTask.md

```diff
@@ -16,9 +16,26 @@ This task aims to provide a fluent and easy to use mechanism for the consumer to
 - The task should signal upload completion via by returning the status or returning a completed **Task**/**Promise**/**Future** for async APIs.
 - Using the RequestContext, the feature flag for the **FileUploadTask** can be set for telemetry purposes.
 - The task should not retry to upload failed slices as any retry should already be done by the retry-handler which the task should be using.
-- The task should provide cancellation capabilities through native task cancellation sources when using async APIs.
-- For Outlook, OneDrive and PrintDocument APIs, send a delete request to the upload session URL to cancel an upload task. 
+- Upload task cancellation - 
+  - The task should provide cancellation capabilities through native task cancellation sources when using async APIs.
+  - For Outlook, OneDrive and PrintDocument APIs, send a delete request to the upload session URL to cancel an upload task. 
 - The task should provide api to access the upload session object containing its URL, expiry date and the cancellation status.
+  - Implement `getUploadSession()` which returns a `LargeFileTaskUploadSession` object.
+  - `LargeFileTaskUploadSession` -
+    ```json
+    "properties": {
+      "url": { "type": "string"},
+      "expiryDate": { "type": "date"},
+      "isCancelled": { "type": "boolean"},
+    }
+    ```
+- Progress Handler -
+  - Provide capabilities to the user to track progress using callback functions.
+  - Implement a `IProgress` interface which contains the following callback properties -
+    - `progress(rangeOfFileUploaded, additionalParamters):void` - called after each chunk upload.
+    - `completed(uploadResult, additionalParamters):void` - called on a successful task completion.
```

Comment:
the completion and the failure can be signaled by the completion or the failure of the Task/CompletableFuture/Promise in most languages. I don't think we need to redefine those. As for the additional parameters, I think it's a nice addition for the progress one, they'll already have access to those from the context and/or by chaining the Task/CF/Promise.

Reply:
Removed `completed` and `failure.` I have renamed the interface to `IUploadEventHandler` keeping this open for future events other than `progress`.